push-build.sh

```diff
@@ -1348,6 +1348,29 @@ release::gcs::publish () {
   fi
 }
 
+ensure_crane_exists () {
+  if ! command -v crane &> /dev/null; then
+    echo "Installing crane..."
+    CRANE_VERSION="v0.19.1"
+    BASE_URL="https://github.com/google/go-containerregistry/releases/download"
+    TAR_NAME="go-containerregistry_Linux_x86_64.tar.gz"
+
+    curl -sL "${BASE_URL}/${CRANE_VERSION}/${TAR_NAME}" > go-containerregistry.tar.gz
```

Comment:
Does it make sense to use `$TMPDIR` for storing the tarball? Also, should we use `--retry` option in case of network hiccups?

Reply:
Some more nits:
- should we use `-o` instead of redirecting the output?
- should we use `--fail` option to fail if request failed?

Reply:
Added ``$TMPDIR``, ``--retry``, ``-o``. and ``--fail``.